migrate/20230711_add_nic.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  change do
+    rename_table(:vm_private_subnet, :private_subnet)
+
+    alter_table(:private_subnet) do
+      drop_foreign_key :vm_id
+      add_column :state, :text, null: false, default: "creating"
+    end
+
+    create_table(:nic) do
+      column :id, :uuid, primary_key: true, default: Sequel.lit("gen_random_uuid()")
+      column :tap_name, :text, null: false, collate: '"C"'
+      foreign_key :private_subnet_id, :private_subnet, primary_key: true, type: :uuid
+      column :mac, :text, null: false
+      column :created_at, :timestamptz, null: false, default: Sequel.lit("now()")
+      column :private_ipv4, :inet, null: false
+      column :private_ipv6, :inet, null: false
+      column :vm_id, :uuid, null: false
+      column :encryption_key, :text, null: false
+    end
+
+    alter_table(:vm) do
+      add_foreign_key :nic_id, :nic, type: :uuid
+      add_foreign_key :private_subnet_id, :private_subnet, type: :uuid
```

Comment:
isn't this duplicate/conflicts with multiple NICs in multiple subnets? seems like you might want neither column to do something like `Nic.where(vm_id: Vm.id)` (but Sequel's `many_to_one` association code should handle it for you)